src/TLuaInterpreter.cpp

```diff
@@ -10217,86 +10217,151 @@ int TLuaInterpreter::getCmdLine(lua_State* L)
 
 int TLuaInterpreter::installPackage(lua_State* L)
 {
-    string event;
+    QString packageName;
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "installPackage(): wrong argument type");
-        lua_error(L);
-        return 1;
+        lua_pushfstring(L, "installPackage: bad argument #1 type (pathFileName of package as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error(L);
     } else {
-        event = lua_tostring(L, 1);
+        packageName = QString::fromUtf8(lua_tostring(L, 1));
+        if (packageName.isEmpty()) {
+            lua_pushnil(L);
+            lua_pushstring(L, "An empty string is not a valid pathFileName for a package to install.");
+            return 2;
+        }
     }
+
     Host& host = getHostFromLua(L);
-    QString package = event.c_str();
-    host.installPackage(package, 0);
-    return 0;
+    QString errorMessage;
+    if (host.installPackage(packageName, 0, &errorMessage)) {
+        lua_pushboolean(L, true);
+        return 1;
+    } else {
+        lua_pushnil(L);
+        lua_pushfstring(L, "Failed to install package \"%s\", error message: %s.",
```

Comment:
Hm, well it obviously failed because it didn't return `true`. Simply return the error message itself.

Reply:
Well it does fit the __nil+error message__ format and does hint what caused the appended message to come from, without checking further I can't say what the second `%s` will look like.

\<aside>I do wonder whether we will have fun when we turn on I18n/L10n that error messages generated by the Qt and other libraries will be in the user selected locale/language...!\</aside>

Reply:
Ooops, looking more carefully I see that the five of the six possible "error messages:" that could be returned as `errorMessage` were of the form `[ ERROR ] - blah, blah blah: error text from libzip.` (and the other, sixth, is the same without a libzip message) and that isn't right for here!  It _seems_ as though now this is the only use of `mudlet::unzip` - which is wierd as I was sure there were other places that zip files were processed...